hudi-common/src/main/java/org/apache/hudi/common/model/HoodieRecordPayload.java

```diff
@@ -44,6 +44,22 @@
   @PublicAPIMethod(maturity = ApiMaturityLevel.STABLE)
   T preCombine(T another);
 
+  /**
+   * When more than one HoodieRecord have the same HoodieKey, this function combines all fields(which is not null)
```

Comment:
please keep this javadoc to be just about the preCombine() method, without any context from this PR;s scenarios.

Reply:
We can have this javadoc in the actual impl